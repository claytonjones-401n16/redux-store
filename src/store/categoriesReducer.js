export const initState = {
  allCategories: [],
  currentCategory: '',
};

export default (state = initState, action) => {
  let newState = { ...state };

  switch (action.type) {
      case 'GET_CATEGORIES':
        newState.allCategories = [...action.payload];
        break;
      case 'CHANGE_CATEGORY':
          newState.currentCategory = action.payload;
          break;
      case 'ADD_CATEGORY':
          newState.allCategories.push(action.payload);
          break;
      default:
          break;
  }

  return newState;
};

