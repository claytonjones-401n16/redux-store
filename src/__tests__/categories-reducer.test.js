import categoriesReducer, {initState} from '../store/categoriesReducer';

describe('categories reducer', () => {
  it('should return the initial state', () => {
    expect(categoriesReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle GET_CATEGORIES', () => {
    expect(categoriesReducer(initState,
      {
        type: 'GET_CATEGORIES',
        payload: ['cat1', 'cat2', 'cat3'],
      })
    ).toMatchSnapshot();
  });

  it('should handle CHANGE_CATEGORY', () => {
    expect(categoriesReducer(initState,
      {
        type: 'CHANGE_CATEGORY',
        payload: 'cat1',
      })
    ).toMatchSnapshot();
  });

});