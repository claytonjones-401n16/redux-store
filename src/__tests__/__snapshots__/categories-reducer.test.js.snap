// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categories reducer should handle CHANGE_CATEGORY 1`] = `
Object {
  "allCategories": Array [],
  "currentCategory": "cat1",
}
`;

exports[`categories reducer should handle GET_CATEGORIES 1`] = `
Object {
  "allCategories": Array [
    "cat1",
    "cat2",
    "cat3",
  ],
  "currentCategory": "",
}
`;

exports[`categories reducer should return the initial state 1`] = `
Object {
  "allCategories": Array [],
  "currentCategory": "",
}
`;
